File: code/stack.py
class Stack():
    def __init__(self):
        self.counter = 0
        self.storage = {}

    def push(self, data):
        self.storage[self.counter] = data
        self.counter += 1

    def size(self):
        return self.counter

    def pop(self):
        if self.counter == 0:
            return
        self.counter -= 1
        del self.storage[self.counter]
    
    def is_empty(self):
        return len(self.storage) == 0

    def peek(self):
        return self.storage[self.counter-1]

    def print_elements(self):
        for i in self.storage:
            print(self.storage[i])




stack = Stack()
stack.push('A')
stack.push('B')
stack.push('C')
stack.pop()
stack.print_elements()
print('Size:', stack.size())
print(stack.is_empty())
print(stack.peek())

File: code/stack.js
class Stack {
    constructor() {
        this.counter = 0
        this.storage = {}
    }

    push(data) {
        this.storage[this.counter] = data
        this.counter++
    }

    pop() {
        if (this.counter === 0) return undefined
        delete this.storage[this.counter - 1]
        this.counter--
    }

    size() {
        return this.counter
    }

    print() {
        for (let i = 0; i < this.counter; i++) {
            console.log(this.storage[i])
        }
    }

}


const stack = new Stack()
stack.push('1')
stack.push('12')
stack.push('13')
stack.push('14')
stack.pop()
stack.print()
stack.size()